library/general/src/lib/ui/text_helpers.rb

```diff
@@ -20,36 +20,48 @@
 module UI
   # Provides a set of methods to manipulate and transform UI text
   module TextHelpers
-    # Wrap given text breaking lines longer than given wrap size. It supports
-    # custom separator, max number of lines to split in and cut text to add
-    # as last line if cut was needed.
+    # Wrap text breaking lines in the first whitespace that does not exceed given line width
     #
-    # @param [String] text to be wrapped
-    # @param [String] wrap size
-    # @param [Hash <String>] optional parameters as separator and prepend_text.
-    # @return [String] wrap text
-    def wrap_text(text, wrap = 76, separator: " ", prepend_text: "",
-      n_lines: nil, cut_text: nil)
-      lines = []
-      message_line = prepend_text
-      text.split(/\s+/).each_with_index do |t, i|
-        if !message_line.empty? && "#{message_line}#{t}".size > wrap
-          lines << message_line
-          message_line = ""
-        end
+    # Additionally, it also allows retrieving only an excerpt of the wrapped text according to the
+    # maximum number of lines indicated, adding one more with the cut_text text when it is given.
+    #
+    # @param text [String] text to be wrapped
+    # @param line_width [Integer] max line length
+    # @option preprend_text [String] the text to prepended
+    # @param n_lines [Integer, nil] the maximum number of lines
+    # @param cut_text [String] the omission text to be used when the text should be cut
+    #
+    # @return [String]
+    def wrap_text(text, line_width = 76, prepend_text: "", n_lines: nil, cut_text: "")
+      full_text = prepend_text + text
 
-        message_line << separator if !message_line.empty? && i != 0
-        message_line << t
+      return full_text if line_width > full_text.length
+
+      wrapped_text = full_text.lines.collect! do |line|
+        l = (line.length > line_width) ? line.gsub(/(.{1,#{line_width}})(?:\s+|$)/, "\\1\n") : line
+        l.strip
       end
 
-      lines << message_line if !message_line.empty?
+      result = wrapped_text.join("\n")
+      result = excerpt(result, n_lines, omission: cut_text) if n_lines
+      result
+    end
+
+    # Extract an excerpt of given text if it has more lines than the maximum allowed
+    #
+    # @param text [String]
+    # @param max_lines [Integer]
+    # @param omission [String] the text to be added
+    #
+    # @return [String] an excerpt if given text had more lines than requested; full text otherwise
+    def excerpt(text, max_lines, omission: "")
```

Comment:
name is really hard to remember. What about max_lines method name?

Reply:
What do you think about `#fragment` or `#passage` (https://www.thesaurus.com/browse/excerpt)

Reply:
well, my issue is that it is not real fragment or passage. It is just limiting text to have at max X lines. You cannot have senteces or stuff like this, so it can looks like it can do more then it really do.

Reply:
Ah! Now I got it :)

Ok, what about [`head`](https://github.com/yast/yast-yast2/pull/980/commits/ea2d2d304b234721aae82b91547a1593f50fee68)? Just a suggestion. I can change it to `max_lines` if you think it's not a good idea to use that word in a helper or still being not completely accurate.

Thanks!

Reply:
head is fine for me.